subprojects/core/src/main/java/org/gradle/cache/internal/DefaultCacheAccess.java

```diff
@@ -499,4 +511,80 @@ Thread getOwner() {
     FileAccess getFileAccess() {
         return fileAccess;
     }
+
+    protected static class IndexedCacheEntry {
+        private final MultiProcessSafePersistentIndexedCache cache;
+        private final PersistentIndexedCacheParameters parameters;
+
+        public IndexedCacheEntry(PersistentIndexedCacheParameters parameters, MultiProcessSafePersistentIndexedCache cache) {
+            this.parameters = parameters;
+            this.cache = cache;
+        }
+
+        public MultiProcessSafePersistentIndexedCache getCache() {
+            return cache;
+        }
+
+        public PersistentIndexedCacheParameters getParameters() {
+            return parameters;
+        }
+
+        public void assertCompatibleCacheParameters(PersistentIndexedCacheParameters parameters) {
+            List<String> faultMessages = new ArrayList<String>();
+
+            assertCacheNameMatch(faultMessages, parameters.getCacheName());
```

Comment:
nit, `assert` gives off the wrong impression (that this could cause an exception/assertion error).  maybe `check`?